### PR TITLE
ref(streams): Add `headers` to `KafkaPayload`

### PR DESCRIPTION
**Pull request by tkaemming**

### CODE REVIEW

File: snuba/utils/streams/kafka.py [hunk 1]
```diff
@@ -58,12 +58,30 @@ def __init__(self, state: KafkaConsumerState):
         self.__state = state
 
 
-@dataclass(frozen=True)
+Headers = Sequence[Tuple[str, bytes]]
```
Comment (tkaemming) [inline snuba/utils/streams/kafka.py]: This maps to the return value of `Message.headers()` as documented here: https://docs.confluent.io/current/clients/confluent-kafka-python/#confluent_kafka.Message.headers

More detail here on the general design here: https://cwiki.apache.org/confluence/display/KAFKA/KIP-82+-+Add+Record+Headers

File: snuba/utils/streams/kafka.py [hunk 2]
```diff
@@ -58,12 +58,30 @@ def __init__(self, state: KafkaConsumerState):
         self.__state = state
 
 
-@dataclass(frozen=True)
```
Comment (tkaemming) [inline snuba/utils/streams/kafka.py]: dataclasses don't support `__slots__` with default arguments. 😞 

I think it's more important that this retains a lightweight memory footprint and has a default value than using dataclasses without the default argument.

File: snuba/utils/streams/kafka.py [hunk 3]
```diff
@@ -58,12 +58,30 @@ def __init__(self, state: KafkaConsumerState):
...
+Headers = Sequence[Tuple[str, bytes]]
+
+
 class KafkaPayload:
-    __slots__ = ["key", "value"]
+    __slots__ = ["__key", "__value", "__headers"]
```
Comment (tkaemming) [inline snuba/utils/streams/kafka.py]: I was kind of surprised this works, but it does.

**Review by tkaemming — COMMENTED** [review]
The main motivations here for adding this to the payload instead of the `Message` object:

1. This is something that is sort of particular to Kafka (sort of like message key) so it seemed reasonable to encapsulate it here.
2. I didn't want to make a decision on which message headers to return to maintain compatibility with the `TickConsumer` here: https://github.com/getsentry/snuba/blob/e747a1788cc184ddcd4c0c92bc877d81e6141aa4/snuba/subscriptions/consumer.py#L110-L118

**Review by fpacifici — APPROVED** [review]

**Pull request merged**